cv32e40p/bsp/link.ld

```diff
@@ -242,8 +242,23 @@ SECTIONS
   . = SEGMENT_START("ldata-segment", .);
   . = ALIGN(32 / 8);
   __bss_end = .;
+  /* In toolchain, x3 is the gp global pointer register and gp relaxation occurs.
```

Comment:
Thank you for this detailed comment.  (Did I mention I hate the toolchain?)

Reply:
Toolchain is just implementing what is defined in RISC-V ABI 😀.